### PR TITLE
Bugfix/fix double turn and oil drop

### PR DESCRIPTION
**Pull request by Arne27**
This bug was to address the bugs that were raised here

https://github.com/EntelectChallenge/2020-Overdrive/issues/17
https://github.com/EntelectChallenge/2020-Overdrive/issues/23

### CODE REVIEW

**Comment by styphoiz** [comment]
@Arne27, can you have a look at the boost/oil powerups as well, see my comments here https://forum.entelect.co.za/t/initial-speed-mud-end-tile/900/20

**Comment by Arne27** [comment]
@styphoiz I believe that this PR should resolve all issues regarding powerups/penalties. 

The intention is that you only receive a powerup or penalty once, I am also updating the readme to reflect this as we speak

File: game-engine/src/main/scala/za/co/entelect/challenge/game/contracts/map/CarGameMap.scala [hunk 1]
```diff
@@ -291,4 +289,9 @@ class CarGameMap(players: util.List[Player], mapGenerationSeed: Int, lanes: Int,
       carGamePlayer.finish();
     }
   }
+
+  private def findPositionToStartCountingColisionsFrom(oldPosition: BlockPosition): BlockPosition = {
+    val positionToStartCountingMud: BlockPosition = new BlockPosition(oldPosition.getLane(), oldPosition.getBlockNumber() + 1)
```
Comment (demaniak) [inline game-engine/src/main/scala/za/co/entelect/challenge/game/contracts/map/CarGameMap.scala:294]: @Arne27 - maybe stupid question, but does this work correctly when turning ?

Comment (Arne27) [reply game-engine/src/main/scala/za/co/entelect/challenge/game/contracts/map/CarGameMap.scala:294]: It should not affect the turning mechanism. I will check though, if you run into any issues please don’t hesitate to raise it

**Pull request merged**